libcudacxx/include/cuda/std/detail/libcxx/include/__type_traits/promote.h

```diff
@@ -19,6 +19,11 @@
 #include "../__utility/declval.h"
 #include "../cstddef"
 
+#if defined(__cuda_std__) && defined(_LIBCUDACXX_CUDACC)
```

Comment:
What is this actually guarding against? Do we want to exclude nvrtc? or also clang-cuda

We should introduce a `_LIBCUDACXX_HAS_EXTENDED_FLOATING_TYPES` define or something along the way